pkg/gitparse/gitparse.go

```diff
@@ -373,16 +385,21 @@ func (c *Parser) FromReader(ctx context.Context, stdOut io.Reader, diffChan chan
 			latestState = MergeLine
 		case isAuthorLine(isStaged, latestState, line):
 			latestState = AuthorLine
-			currentCommit.Author = strings.TrimRight(string(line[8:]), "\n")
-
-		case isDateLine(isStaged, latestState, line):
-			latestState = DateLine
+			currentCommit.Author = strings.TrimSpace(string(line[8:]))
+		case isAuthorDateLine(isStaged, latestState, line):
+			latestState = AuthorDateLine
 
-			date, err := time.Parse(c.dateFormat, strings.TrimSpace(string(line[6:])))
+			date, err := time.Parse(c.dateFormat, strings.TrimSpace(string(line[12:])))
 			if err != nil {
-				ctx.Logger().V(2).Info("Could not parse date from git stream.", "error", err)
+				ctx.Logger().Error(err, "failed to parse commit date", "commit", currentCommit.Hash)
 			}
 			currentCommit.Date = date
+		case isCommitterLine(isStaged, latestState, line):
+			latestState = CommitterLine
+			currentCommit.Committer = strings.TrimSpace(string(line[8:]))
+		case isCommitterDateLine(isStaged, latestState, line):
+			latestState = CommitterDateLine
```

Comment:
Is there any particular reason you called the state "Commit**ter**DateLine" when the Git output is "CommitDate"?

Reply:
It's to disambiguate the committ**er** information:
```
Commit: Dustin Decker <dustin@trufflesec.com>
CommitDate: Tue Feb 6 10:13:53 2024 -0500
```

With the commit information:
```
commit 4218c39d99b5f30153f62471c1be1c1596f0a4d4
```

I don't know why they have `Author` and `Commit` instead of `Committer`, when committer is used explicitly in the docs. Perhaps they wanted to keep the line lengths the same?

Reply:
that's...very strange. but ok!

Reply:
¯\\_(ツ)_/¯